src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -197,13 +197,6 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
             SyntaxKind kind,
             CSharpSyntaxNode syntax)
         {
-            Debug.Assert(
```

Comment:
Why remove the assertion?

Reply:
`ToDeclarationKind` one line below will fire an exception in all instances if the assert isn't followed, so this just feels like duplication